brave/brave-core
# Remove content script restrictions on Uphold

Fixes brave/brave-browser#6309 by removing the content script half of https://github.com/brave/brave-core/pull/2946.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

The webRequest hack can manually tested using https://github.com/fmarier/webrequest-demo:

1. Install the extension via developer mode in `chrome://extensions`.
2. Visit https://sandbox.uphold.com/something and ensure that you're redirected to `fmarier.org` with the original URL in the query string.
3. Visit https://sandbox.uphold.com/authorize/blah and ensure that you don't get redirected to `fmarier.org` until Uphold redirects you to `https://sandbox.uphold.com/authorize/404` first.
4. Visit https://api.uphold.com/oauth2/token and ensure that you're not redirected to `fmarier.org`.
5. Remove the extension.

The contentScript hack can be tested using https://github.com/fmarier/contentscript-demo:

1. Install the extension via developer mode in `chrome://extensions`.
2. Visit https://sandbox.uphold.com/something and ensure that you see a popup window.
3. Visit https://sandbox.uphold.com/authorize/blah and ensure that you get a popup window before and after Uphold redirects you to `https://sandbox.uphold.com/authorize/404`.
4. Visit https://api.uphold.com/oauth2/token and ensure that you see a popup window.
5. Repeat steps 2 and 3 with `uphold.com` instead of `sandbox.uphold.com`.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
